front/lib/api/assistant/agent.ts

```diff
@@ -292,6 +295,29 @@ async function* runMultiActionsAgentLoop(
   }
 }
 
+async function getEmulatedAgentMessageActions(
+  auth: Authenticator,
+  {
+    agentMessage,
+    conversation,
+  }: { agentMessage: AgentMessageType; conversation: ConversationType }
+): Promise<AgentActionType[]> {
+  const actions: AgentActionType[] = [];
+  if (await isJITActionsEnabled(auth)) {
+    const a = makeConversationListFilesAction(agentMessage, conversation);
+    if (a) {
+      actions.push(a);
+    }
+  }
+
+  // We ensure that all emulated actions are injected with step -1.
+  assert(
+    actions.every((a) => a.step === -1),
```

Comment:
can't this also be enforced in type system removing the need for assert ?

Reply:
That would mix their type and the fact that they are emulated. One could imagine that a classic action is emulated.

Reply:
Fair !

Reply:
@spolu

nit: the -1 index is going to be limiting if we need to order multiple emulated actions, I suggest < 0